### ISSUE TITLE
Cannot deregister volume while its id is a prefix of other volume id.

### ISSUE THREAD

**Issue by kriestof** [hashicorp/nomad#12005]
### Nomad version
1.2.5

### Operating system and Environment details
Arch linux

### Issue

I try to deregister volume which id is a prefix of other volume. But then I got an error

```
nomad volume deregister registry-volume
Prefix matched multiple volumes

ID                Name             Plugin ID  Schedulable  Access Mode
registry-volume   Docker registry  ceph-csi   true         multi-node-multi-writer
registry-volume1  Docker registry  ceph-csi   true         multi-node-multi-writer
```

It's non-critical and definitely a corner case as I can deregister registry-volume1 and then registry-volume.

### Reproduction steps

Register 2 volumes registry-volume and registry-volume1. Then try to deregister registry-volume.

#### Expected Result
Volume matched by exact id should be deregistered.

#### Actual Result
There is an error that multiple prefixes was matched.

**Comment by tgross** [comment]
Hi @kriestof! We fixed a bug in the deregister prefix matching in https://github.com/hashicorp/nomad/pull/11852 which shipped in Nomad 1.2.4, where prefixes were used to deregister. But you're right, we should also have followed that up with changes to the CLI to account for exact matches, similarly to how we do for job ID.

**Comment by tgross** [comment]
Fixed in https://github.com/hashicorp/nomad/pull/12051 which should ship in Nomad 1.3.0
